### PR TITLE
fix builtins plugins from leaking vars

### LINKED ISSUE #461: minify-builtins leaks vars
minify-builtins leaks the variables it produces. It needs to look and see if the rest of the code is in an IIFE and put its vars inside if so. And being outside the IIFE, the vars aren't mangled (#425) so that it doesn't decrease the file size and probably not even the gzipped size!

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
fix #461 
+ The implementation decides where to insert the created identifier based on the input code. 

Even after fixing this, the mangler doesn't rename the identifier properly. Related - https://github.com/babel/babili/issues/425

### CODE REVIEW

**Comment by curiousdannii** [comment]
If I'm reading the tests correctly this still leaks some vars. Unless it can detect it's in module mode it would be better not to try to convert top level builtins.

File: packages/babel-plugin-minify-builtins/src/index.js [hunk 1]
```diff
@@ -123,6 +124,26 @@ module.exports = function({ types: t }) {
...
 
+// decides the best path to insert the newly created identifier
+function getPath(program, firstPath, lastPath) {
+  let resultPath = program;
+  // check if occurence of member exp ex - Max.max() is before the
+  // block statement container
+  const { start: nodeStart } = firstPath.parent;
+  const { end: nodeEnd } = lastPath.parent;
+
+  program.traverse({
```
Comment (boopathi) [inline packages/babel-plugin-minify-builtins/src/index.js]: the same can be achieved by `getFunctionParent` recursively till `program`. `program.traverse` inside a loop will be costly.

**Comment by vigneshshanmugam** [comment]
[Update]

Running mangler on exit fixes the #425 issue.

**Comment by vigneshshanmugam** [comment]
Works now for functions at same level.

**Pull request merged**